internal/featuregates/mapper.go

```diff
@@ -0,0 +1,60 @@
+package featuregates
+
+import (
+	"bytes"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/api/features"
+)
+
+const (
+	// Add your new upstream gate here
+	ForceSemverUpgradeConstraintsGate = "ForceSemverUpgradeConstraints"
+)
+
+type MapperInterface interface {
+	UpstreamForDownstream(downstreamGate configv1.FeatureGateName) ([]string, error)
+}
+
+// Mapper knows the mapping between downstream and upstream feature gates
+type Mapper struct {
+	gates map[configv1.FeatureGateName][]string
+}
+
+func NewMapper() *Mapper {
+	// Add your downstream to upstream mapping here
+	gates := map[configv1.FeatureGateName][]string{
+		features.FeatureGateNewOLM: {ForceSemverUpgradeConstraintsGate},
+	}
+
```

Comment:
For now, I think we should loop through the configured gates and `panic` if:
- the gate name is `NewOLM` (we don't want to accidentally turn a non-default-enabled feature gate on in production.
- they don't have a `NewOLM` prefix (we want all sub features of OLMv1 to be "under" `NewOLM` which is the parent feature gate for _all_ of OLMv1.

We should probably explain this in the panic message, and then maybe include a comment that suggests that relaxing these rules may be necessary if we ever decide we want to enable an alpha or beta feature from upstream in OCP GA downstream, or if a completely separate OCP feature impacts how we would deploy our components.

Reply:
I've added panic checks and comments. I thought about just leaving the `NewOLM` as an empty mapping with a comment, but better to have an additional verification for now at least and it costs us nothing.

Regarding the 'NewOLM' prefix - I've added panic check as well, but if it ever triggers it would mean that a non-prefixed flag has already passed review and has been added to `github.com/openshift/api`, thinking out loud here, it would be good to be able to place a failsafe like that directly in the api.

Reply:
There are a bunch of other features in openshift/api that are completely unrelated to `NewOLM` and OLMv1. So my thinking here is that this is our way of saying "any OLMv1-specific features _MUST_ be prefixed with `NewOLM`".

I agree, it would be better to have that failsafe in `openshift/api`, but I'm not sure how realistic that is given all the other features for other OCP components that are registered there.